### ISSUE TITLE
Form factors viz

### ISSUE THREAD

**Issue by ShaineRosewel** [Devographics/Monorepo#257]
Yesterday, I shared this plot but this only includes answer sets with high responses.

fig 1:
<img src="https://github.com/Devographics/Monorepo/assets/62248526/850e1178-91f8-40bb-b3d2-b3afb5bcb6de" alt="FFT" width="600">

For the question, is there anyone who tests on mobile alone, @atopal said it may not be likely because people dev on desktop. But I tried to figure that out using another plot (fig 3). I also recalled @SachaG mentioned it might be of value to present the number of options respondents ticked such as the plot below (fig 2). 

fig 2: 
<img src="https://github.com/Devographics/Monorepo/assets/62248526/1ef202ee-192d-41f9-8900-34b4a2b6408b" alt="FF" width="600">

This plot below shows top 10 form factor combination for each number of form factors ticked (I only included 1 to 5 ticked answers) - this combines fig 2 and fig 1 into one chart.

fig 3:
![faceted](https://github.com/Devographics/Monorepo/assets/62248526/416fa023-c7b1-4a99-a5e6-1d8578c7aebb)

Right now, what we have is this

<img src="https://github.com/Devographics/Monorepo/assets/62248526/0976a5e4-85eb-42f7-88c1-78e8f36dbc81" width="650">

The question for this item is **Which form factors or environments do you test on?** Not sure if we can reword this to ask, Across which form factors do you usually test on? So that when we present the results, the combinations will be meaningful - we get to know that people usually want to make sure their apps work both on desktop and smartphone - not sure though if that is what you want @SachaG. I also am not sure how respondents interpret the question. Because I am thinking apps can be tested on one device, but can also be across devices. 

@michaelquiapos, @Gio-Q

**Comment by SachaG** [comment]
For now I am not too worried about this specific question, but I am more interested in how we could apply this kind of grouped analysis to all questions. Could you share more information about what algorithm or function you used to create it?

**Comment by ShaineRosewel** [comment]
Seems like I can't see a similar one with nivo. But there must be a way to customize, I used ggplot2 package in R, the function that made that possible is facet_wrap.

**Comment by SachaG** [comment]
Nivo is mostly a visualization library, so it doesn't really do any calculations as far as I know.

**Comment by ShaineRosewel** [comment]
As for the calculations, It is similar to the grouped version. But in R, I have the option to use facets over the grouped format. I am thinking it is more of a layout than calculation concern...

**Comment by ShaineRosewel** [comment]
By defining groups, do you mean, something similar to high and low income group - which was formed from the original choices?

**Comment by SachaG** [comment]
No I mean like `desktop smartphone`, `desktop feature phone`, etc. I was wondering if you just tested every combination of 2, 3, 4, 5, etc. items or if there was some kind of selection algorithm to build the pairs/triplets/etc. .

**Comment by ShaineRosewel** [comment]
Nope, those were the combinations from the respondents - things they checked together.

**Comment by SachaG** [comment]
What I mean is that if we have a list with 10 items then we would have

- 10 1-item combinations
- 45 2-item combinations
- 120 3-item combinations
- 210 4-item combinations
- etc. 

In some cases we might have 30 or even 50 options. With 30 options, there are 4060 3-item combinations and 27405 4-item combinations. So if we're just testing all of them, that doesn't seem like it would scale… that's why I was wondering if there was any "smarter" algorithm.

**Comment by ShaineRosewel** [comment]
In the current case, I am not sure how many options we have. But, people only checked up to a maximum of 12. The question is do we include visualization for each of those - from 1 up to 12? It depends, especially that, as this number increases, we are sure that less people have similar answers. Say for example if i test on both mobile and desktop, i am sure many people do the same. But if i choose 12 of those, it is likely that other respondents who chose 12 as well chose a different set. We can present up to a combination size that give insightful results. This means we do have to check if the resulting plot delivers valuable insight.


<img width="629" alt="Screenshot 2023-08-16 at 6 07 23 PM" src="https://github.com/Devographics/Monorepo/assets/62248526/9bac88d4-99ec-4770-8843-34e6de53b8d2">


I want to believe that your current version of plot is all good but I think it will add value if we present at least fig 1. It is not made to show the top responses for each combination size, but gives an idea that many users choose say both desktop and smartphone. This in fact surpasses the number of respondents who only checked desktop (which is the top on your current plot). This means that respondents mostly create apps that are tested on both desktop and mobile.

**Comment by SachaG** [comment]
Yes I think in any case we can limit ourselves to triplets. I will keep thinking about the best way to visualize all this…
